src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -192,7 +199,7 @@ SafeToPushdownUnionSubquery(RelationRestrictionContext *restrictionContext)
 			/* union does not have partition key in the target list */
 			if (partitionKeyIndex == 0)
 			{
-				return false;
+				continue;
```

Comment:
We'd probably want to remove the above check `if (PartitionMethod(relationId) == DISTRIBUTE_BY_NONE)`, but not in this PR I guess?